architecture/protocol.go

```diff
@@ -147,6 +147,16 @@ func (command *Command) Parse(rawCommand string) (bool, error) {
 			return true, command.Err
 		}
 
+		// Check if parameter types are correct
+		for i, _ := range opts.Params {
+			if opts.ParamInteger[i] {
+				_, err := strconv.Atoi(parts[i+1])
```

Comment:
@marekyggdrasil check out #77 for a solution to parsing/validating just once.  @vimukthi-git suggested using `interface{}` instead of adding everything to `Command`, which is doable.

I'm going to work on #39 after lunch.  Do you want to take a crack at refactoring #77--or incorporating it into this PR?  If not, I'll handle this after #39.